3515/2/commands/checkin.rb

```diff
@@ -0,0 +1,85 @@
+require_relative '../helper_modules/directory_maker'
+require_relative '../helper_modules/file_write'
+module CheckinCommand
+  def checkin!(*)
+    if user_authorized?
+      checkin_false?
+    else
+      respond_with :message, text: "Извините, но ваш аккаунт не авторизован\nДля авторизации введите
+      команду /start"
+    end
+  end
+
+  def user_authorized?
+    !(User[from['id']]).to_s.empty?
+  end
+
+  def checkin_false?
+    if User[from['id']].checkin == 'false'
+      photo_request_for_checkin
+    else
+      respond_with :message, text: "Ошибка. На данный момент вы уже зачекинились в лагере\n
+      Используйте команду /checkout"
+    end
+  end
+
+  def photo_request_for_checkin
+    respond_with :message, text: 'Отправь мне селфи :)'
+    save_context :check_api_photo_file_path_for_checkin
+  end
+
+  def check_api_photo_file_path_for_checkin(*)
+    if payload.include?('photo')
+      respond_with :message, text: 'Получил фото. Сохраняю...'
+      api_photo_file_true_for_checkin(payload)
+    else
+      respond_with :message, text: 'Ну я же вежливо попросил... '
+      photo_request_for_checkin
+    end
+  end
+
+  def api_photo_file_true_for_checkin(payload)
+    photo_id = payload['photo'].last['file_id']
+    uri = URI(ENV['API_URL'] + ENV['TOKEN'] +
+              '/getFile?file_id=' + photo_id)
+    photo_path = JSON.parse(Net::HTTP.get(uri))['result']['file_path']
+    photo_downloader_for_checkin(photo_path)
+  end
+
+  def photo_downloader_for_checkin(photo_path)
+    checkin_path = "public/#{from['id']}/checkin/#{Time.now}"
+    DirectoryMaker.dir_maker(checkin_path)
+    photo_path_new = "#{checkin_path}/selfie.jpg"
+    FileWrite.file_write(photo_path_new, URI(ENV['API_URL_FILE'] + ENV['TOKEN'] + '/' + photo_path))
+    request_geolocation_for_checkin
+  end
+
+  def request_geolocation_for_checkin
+    respond_with :message, text: 'А теперь отправь мне геолокацию...'
+    save_context :get_geolocation_for_checkin
+  end
+
+  def get_geolocation_for_checkin(*)
+    if payload.include?('location')
+      respond_with :message, text: 'Получил геолокацию. Проверяю...'
+      geolocation = payload['location'].values.to_a
+      geolocation_checker_for_checkin(geolocation)
+    # binding.pry
+    else
+      respond_with :message, text: 'Ну я же вежливо попросил...'
+      request_geolocation_for_checkin
+    end
+  end
+
+  def geolocation_checker_for_checkin(geolocation)
+    if geolocation[0].between?(53.9145, 53.916) &&
+       geolocation[1].between?(27.567, 27.57)
+      User[from['id']].set(:checkin, 'true')
+      respond_with :photo, photo: File.open('commands/checkin.jpg')
```

Comment:
I'd recommend to place an empty line before the line(s) of code that return value from a method to improve its readability